lib/javascript/JavascriptParser.js

```diff
@@ -2590,47 +2594,48 @@ class JavascriptParser extends Parser {
 			this.preWalkStatement(statement.declaration);
 			this.prevStatement = prev;
 			this.blockPreWalkStatement(statement.declaration);
+
 			let index = 0;
 			this.enterDeclaration(statement.declaration, def => {
+				// Avoid redundant exports
 				this.hooks.exportSpecifier.call(statement, def, def, index++);
 			});
 		}
+
+		// Process specifiers
 		if (statement.specifiers) {
+			let specifierIndex;
 			for (
-				let specifierIndex = 0;
+				specifierIndex = 0;
 				specifierIndex < statement.specifiers.length;
 				specifierIndex++
 			) {
 				const specifier = statement.specifiers[specifierIndex];
-				switch (specifier.type) {
-					case "ExportSpecifier": {
-						const localName =
-							/** @type {Identifier} */ (specifier.local).name ||
-							/** @type {string} */ (
-								/** @type {Literal} */ (specifier.local).value
-							);
-						const name =
-							/** @type {Identifier} */
-							(specifier.exported).name ||
-							/** @type {string} */
-							(/** @type {Literal} */ (specifier.exported).value);
-						if (source) {
-							this.hooks.exportImportSpecifier.call(
-								statement,
-								source,
-								localName,
-								name,
-								specifierIndex
-							);
-						} else {
-							this.hooks.exportSpecifier.call(
-								statement,
-								localName,
-								name,
-								specifierIndex
-							);
-						}
-						break;
+
```

Comment:
Do you have perf numbers on this optimization?. I'm not aware that moving this from a switch makes a performance difference.

Could you please provide a cpuprofile where you identified the need for the change (in all)